server/src/main/java/org/elasticsearch/index/mapper/MappingParser.java

```diff
@@ -22,34 +20,58 @@
 import java.util.function.Function;
 import java.util.function.Supplier;
 
-public class DocumentMapperParser {
-    private final IndexSettings indexSettings;
-    private final IndexAnalyzers indexAnalyzers;
-    private final Function<String, String> documentTypeResolver;
-    private final DocumentParser documentParser;
-    private final Supplier<Map<Class<? extends MetadataFieldMapper>, MetadataFieldMapper>> metadataMappersSupplier;
+/**
+ * Parser for {@link Mapping} provided in {@link CompressedXContent} format
+ */
+public final class MappingParser {
     private final Supplier<Mapper.TypeParser.ParserContext> parserContextSupplier;
     private final RootObjectMapper.TypeParser rootObjectTypeParser = new RootObjectMapper.TypeParser();
+    private final Supplier<Map<Class<? extends MetadataFieldMapper>, MetadataFieldMapper>> metadataMappersSupplier;
     private final Map<String, MetadataFieldMapper.TypeParser> rootTypeParsers;
```

Comment:
I know they were already called this, but can we rename this map to `metadataTypeParsers`? I always get confused when reading this about why we're using these outside the root.